packages/remix-react/components.tsx

```diff
@@ -929,16 +1171,177 @@ export function useFetcher<TData = any>(): FetcherWithComponents<
+          //  - Expose a hidden field with a stable identifier on the fetcher
+          //    like we did for _hasFetcherDoneAnything
+          key: "todo-not-implemented-yet",
+        },
+        data: undefined,
+      };
+      return fetcher;
+    } else {
+      invariant(false, "nope");
+    }
+  }
+
+  if (state === "loading") {
+    if (formMethod && formAction && formEncType && formData) {
+      if (isActionSubmission) {
+        if (data) {
+          // In a loading state but we have data - must be an actionReload
+          let fetcher: FetcherStates["ReloadingAction"] = {
+            state,
+            type: "actionReload",
+            submission: {
+              method: formMethod.toUpperCase() as ActionSubmission["method"],
+              action: formAction,
+              encType: formEncType,
+              formData: formData,
+              // TODO: this is created as a random hash value in useSubmitImpl in
+              // Remix today. We do not have this key in react router as we
+              // flattened submissions down onto the fetcher.  We can't recreate in
+              // this back-compat layer in a stable manner for useFetchers because
+              // we don't have a stable fetcher identity.  So we could:
+              //  - Expose the fetcher key from the router (might make sense if
+              //    we're considering adding useFetcher({ key }) anyway
+              //  - Expose a hidden field with a stable identifier on the fetcher
+              //    like we did for _hasFetcherDoneAnything
+              key: "todo-not-implemented-yet",
+            },
+            data,
+          };
+          return fetcher;
+        } else {
+          let fetcher: FetcherStates["LoadingActionRedirect"] = {
+            state,
+            type: "actionRedirect",
+            submission: {
+              method: formMethod.toUpperCase() as ActionSubmission["method"],
+              action: formAction,
+              encType: formEncType,
+              formData: formData,
+              // TODO???
+              key: "todo-what-is-this?",
+            },
+            data: undefined,
+          };
+          return fetcher;
+        }
+      } else {
+        // The new router fixes a bug in useTransition where the submission
+        // "action" represents the request URL not the state of the <form> in
+        // the DOM.  Back-port it here to maintain behavior, but useNavigation
+        // will fix this bug.
+        let url = new URL(formAction, window.location.origin);
+
+        // This typing override should be safe since this is only running for
+        // GET submissions and over in @remix-run/router we have an invariant
+        // if you have any non-string values in your FormData when we attempt
+        // to convert them to URLSearchParams
+        url.search = new URLSearchParams(
+          formData.entries() as unknown as [string, string][]
+        ).toString();
```

Comment:
This bug is fixed in `useNavigation` but we wanted to keep the bug behavior in `useTransition`